File: 0485. Max Consecutive Ones/Solution.py
from typing import List


class Solution:
    def findMaxConsecutiveOnes_MK1(self, nums: List[int]) -> int:
        return max(map(len, ''.join(map(str, nums)).split('0')))

    def findMaxConsecutiveOnes_MK2(self, nums: List[int]) -> int:
        maximum = count = 0
        for n in nums:
            if n:
                count += 1
            else:
                maximum = max(maximum, count)
                count = 0
        return max(maximum, count)

File: 0485. Max Consecutive Ones/Solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maximum = 0, count = 0;
        for (int n : nums) {
            if (n == 1)
                count++;
            else {
                maximum = Math.max(maximum, count);
                count = 0;
            }
        }
        return Math.max(maximum, count);
    }
}
